nemo/collections/tts/models/magpietts.py

```diff
@@ -888,6 +910,15 @@ def compute_alignment_loss(self, attention_scores, text_lens, audio_lens, dec_co
         )
         return alignment_loss
 
+    def embed_context_text(self, context_text_tokens):
+        if self.legacy_text_conditioning:
+            context_text_tokens = context_text_tokens - self.tokenizer.tokenizer_offsets[self.text_conditioning_tokenizer_name]
+            context_text_embedded = self.context_text_embedding(context_text_tokens)  # (B, L, E)
```

Comment:
It looks like we're having issues with the legacy path specifically since the aggregated tokenizer adds offsets to the tokens, but we kept separate embeddings in the legacy path so it shouldn't have these offsets.
It'll be best to subtract the offset here from the tokens instead of trying to work it in the Aggregated tokenization class or in the dataset

Reply:
Yes. I believe that's how it is now being done --- Aggregated tokenizer and dataset class work the same way, we just subtract the offset right here.